### DISCUSSION TITLE
Processing Images?

### DISCUSSION THREAD

**Discussion by gltchcom** [toutjavascript/Fooocus-Log-Viewer#12] [Q&A]
![image](https://github.com/toutjavascript/Fooocus-Log-Viewer/assets/153979117/6f70a0b7-ab8a-484e-aa5f-28b4e7cb5088)

Sorry if I missed this documented somewhere, I did search... 
I installed this about an hour ago. Is it scanning all the directories, or do I have to manually go to each one to finish scanning so that I can search?

**Comment by toutjavascript** [comment]
Hello

Logs are processed every time you launch the viewer.html
It could take time if you have a lot of generated images or a slow HDD
In my case, with ssd and 12k images, it takes less than 3 seconds to scan all folders and give access to search and calendar...
Can you tell me more about your configuration ?

**Reply by gltchcom** [reply]
I have about 8k images on a very fast SSD. I was unable to search at any point, and then it seems to have crashed on me. Even if I completely close Fooocus and restart it, I now just get a blank page with a link to this github. 

I am running my output directory outside of the Fooocus directory and used your comment to make that work. I made the symlink and open it with: http://localhost:7865/file=C:/StableDiffusion/Fooocus_win64_2-1-831/Fooocus/outputs/viewer.html 

C:/StableDiffusion/Fooocus_win64_2-1-831/Fooocus/outputs is linked to C:/StableDiffusion/outputs/

I was able to see the images and navigate from folder to folder. Everything worked but the search, and now nothing is working.

**Reply by gltchcom** [reply]
I put the output folder back where it was. viewer.html is displaying very briefly that the folder is empty, then just the link to the github. It then constantly shows "All logs and image are not processed". 

All of that is in Chrome. I just tried Edge and it actually loads. Search still doesn't work though and it still says the images are not processed.

**Comment by toutjavascript** [comment]
Ok
I will test with external images folder
symlink is perhaps the reason...

**Reply by gltchcom** [reply]
I removed the symlink and moved the output folder back. still no luck.

**Comment by gltchcom** [comment]
I've done some troubleshooting. I removed all the folders from the output directory but the most recent. The viewer works absolutely fine. So I start putting them back, and everything works great back to 2024-01-13. If I put any folders created earlier than that, I have the problem. But I also noticed something else.

The calendar view is messed up:
![image](https://github.com/toutjavascript/Fooocus-Log-Viewer/assets/153979117/fa53a771-eac4-4e42-86e6-1326c39e67ce)
There is no 1/12 folder in the outputs, and clicking on it doesn't show any files, even though it says there are 126 images in that folder. But, there are 126 images in the 1/13 folder, and those 126 images are displayed if I open 1/13, but the calendar view says there are 109 images. There are 109 images in 1/14. And this goes all the way to present day. There isn't even an entry for the current day, 2/8/24. It shows I have 4 images on 2/7, but, 2/8 actually has the 4 images. If I manually go to that date it shows the 4 images. So something is still really wrong here.

![image](https://github.com/toutjavascript/Fooocus-Log-Viewer/assets/153979117/093706a9-0b38-44d1-a18c-063a24d479e5)

**Reply by toutjavascript** [reply]
Ok, it seems to be linked to the issue open by an other user
Fix is in the post.
But i cannot manage to reproduce the problem
Could you have a look ?
Thanks !

**Reply by gltchcom** [reply]
I just replaced the functions with what he provided. Now I'm back to the original processing error, so I can't open the calendar to test it.
Does the script read anything else in the folder besides the logfile? Sometimes I upscale an image with another program and it will output in there.

**Reply by toutjavascript** [reply]
The scripts only tries to read the log.html file in every YYYY-MM-DD folder from the creation of Fooocus (about 250 days from now)
If the log file exists, the script parses it and stores metadata of images in an array in memory
At the end of the scan, calendar and search buttons are available.
So if you have other log.html files created by an other app than fooocus it could brake something

**Reply by gltchcom** [reply]
I've had no other apps generating log.html files.. I'm really not sure what's going on.. I'll try to dig into the folders and see if I see anything strange.
